core/dbt/clients/system.py

```diff
@@ -135,11 +135,14 @@ def write_file(path: str, contents: str = '') -> bool:
         make_directory(os.path.dirname(path))
         with open(path, 'w', encoding='utf-8') as f:
             f.write(str(contents))
-    except FileNotFoundError as exc:
-        if (
-            os.name == 'nt' and
-            len(path) >= 260
-        ):
+    except Exception as exc:
```

Comment:
can we also log `exc` / the stack trace here? I'm a-ok with making this exception handler more permissive, but think that our future selves will thank us if we log the actual exception :D

Reply:
I'm going to skip logging the stack trace as it's not going to be super informative - it either came from making the directory or writing to a file, both are pretty much the same problem.

Reply:
yeah - fair enough - I just figure at some point in the future we'll find that some different, equally surprising, error is getting caught in this exception handler and we'll want to know what it is :)